### PR TITLE
NEXUS-21940 Feature flag to hide Conan hosted repository

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-21940

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
   </distributionManagement>
 
   <properties>
-    <nxrm-version>3.19.0-01</nxrm-version>
```
Comment (joedragons) [inline pom.xml]: Q: Is this to be updated to an actual build once we release 3.21?  Is there some trigger to help remind (like 3.next)?

Comment (mlukaretkyi) [reply pom.xml]: We do not have any triggers for community formats. I have just wanted to up parent version to the latest

File: nexus-repository-conan/src/test/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipeTest.java [hunk 2]
```diff
@@ -142,4 +151,24 @@ public void canMatchOnConanPackageZipUploadPackage() {
...
+    System.setProperty(HOSTED_ENABLED_PROPERTY, "false");
+    ConanHostedRecipe conanHostedRecipe = new ConanHostedRecipe(new HostedType(), new ConanFormat(), null);
+    assertThat(conanHostedRecipe.isFeatureEnabled(), is(false));
+  }
+
+  @Test
+  public void hostedDisabledNexusConanHostedEnabledIsNotSet() {
+    ConanHostedRecipe conanHostedRecipe = new ConanHostedRecipe(new HostedType(), new ConanFormat(), null);
+    assertThat(conanHostedRecipe.isFeatureEnabled(), is(false));
+  }
```
Comment (proteus1121) [inline nexus-repository-conan/src/test/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipeTest.java]: maybe make sense to add new line in the end of this file

File: nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy [hunk 3]
```diff
@@ -46,6 +47,8 @@ class ConanHostedRecipe
 {
   public static final String NAME = 'conan-hosted'
 
+  public static final String HOSTED_ENABLED_PROPERTY = 'nexus.conan.hosted.enabled'
```
Comment (proteus1121) [inline nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]: If this constant used only in this file in tests make sense to make it private and mark VisibleForTesting annotation?

Comment (mlukaretkyi) [reply nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipe.groovy]: Yes, we can use VisibleForTesting but, it is just constant, so, for what do we need to hide that? I think this case does not need VisibleForTesting

File: nexus-repository-conan/src/test/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipeTest.java (hunk 2 above)
Comment (mlukaretkyi) [reply nexus-repository-conan/src/test/java/org/sonatype/repository/conan/internal/hosted/ConanHostedRecipeTest.java]: Fixed

**Pull request merged**